src/main/java/com/bieganski/ox/GameFlow.java

```diff
@@ -19,32 +21,53 @@
 
         players.add(Symbol.X);
         players.add(Symbol.O);
-// TODO factory method
-        initializeBoard();
     }
 
     private void initializeBoard() {
         board = new Board(boardSize);
         board.addListener(userInterface);
         board.updateListeners();
-        board.addListener(new Judge(boardSize));
+        Judge judge = new Judge(this, boardSize);
+        board.addListener(judge);
+
     }
 
     @Override
     public void run() {
+        initializeBoard();
+
         PositionCalculator positionCalculator = new PositionCalculator(new PositionValidator(), boardSize);
-        Symbol currentSymbol = players.get(0);
-        while (!win) {
-            userInterface.println("Input row and column:");
-            String input = userInterface.askForInput();
-            boolean added = board.addField(new Field(positionCalculator.calculatePosition(input), currentSymbol));
-            if (added) {
-                currentSymbol = changePlayer(currentSymbol);
-            }
+        currentSymbol = players.get(0);
+
+        makeTurn(positionCalculator);
+
+
+    }
+
+    private void makeTurn(PositionCalculator positionCalculator) {
+        askPlayerForInputAndPlaceSymbol(positionCalculator, currentSymbol);
```

Comment:
Nie do końca łapię czemu kalkulowanie pozycji jest potrzebne do pytania użyszkodnika o to gdzie stawia krzyżyk. :/